hepdata/modules/new_inspire_api/views.py

```diff
@@ -0,0 +1,87 @@
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# HEPData is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HEPData; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+from copy import deepcopy
+
+from flask import request, Blueprint, jsonify
+from hepdata.modules.records.utils.common import record_exists
+from hepdata.resilient_requests import resilient_requests
+from hepdata.modules.new_inspire_api.parser import parsed_content_defaults, get_title, get_doi, get_authors, get_type, get_abstract, \
+    get_creation_date, get_arxiv_id, get_collaborations, get_keywords, get_journal_info, get_year, get_subject_area, updated_parsed_content_for_thesis
+
+blueprint = Blueprint('inspire_datasource', __name__, url_prefix='/inspire')
+
+
+def get_inspire_record_information(inspire_rec_id, verbose=False):
+    url = 'http://inspirehep.net/api/literature/{}'.format(inspire_rec_id)
+    if verbose:
+        print('\rLooking up: ' + url)
+    req = resilient_requests('get', url)
+    status = req.status_code
+
+    if status == 200:
+        content = req.json()
+
+        parsed_content = {
+            'title': get_title(content['metadata']),
+            'doi': get_doi(content['metadata']),
+            'authors': get_authors(content['metadata']),
+            'type': get_type(content['metadata']),
+            'abstract': get_abstract(content['metadata']),
+            'creation_date': get_creation_date(content['metadata']),
+            'arxiv_id': get_arxiv_id(content['metadata']),
+            'collaborations': get_collaborations(content['metadata']),
+            'keywords': get_keywords(content['metadata']),
+            'journal_info': get_journal_info(content['metadata']),
+            'year': get_year(content['metadata']),
+            'subject_area': get_subject_area(content['metadata']),
+        }
+
+        if 'thesis' in parsed_content['type'] and 'thesis_info' in content['metadata'].keys():
+            parsed_content = updated_parsed_content_for_thesis(content, parsed_content)
+        elif 'thesis' in parsed_content['type'] and 'thesis_info' not in content['metadata'].keys():
+            parsed_content['dissertation'] = {}
+
+    else:
+        parsed_content = deepcopy(parsed_content_defaults)
+
+    return parsed_content, 'success'
```

Comment:
Looks like we're passing `'success'` as the status in all cases here. Previously the status code was returned if the call to inspire not successful. I don't think this would affect the `attach_information` endpoint as it seems to ignore the status and carry on regardless (is that worth logging as an issue, @GraemeWatt ?) but it would affect the `migrator`.

Reply:
So should I return the status code if it is different from ```200```? I think I had a ```raise_for_status``` call at the beginning but that was not good. The weird thing is that the ```inspire_api``` is expected to return ```success``` even if the inspire record does not exist (which intuitively I'd say should be a fail).

Reply:
Thanks, @alisonrclarke, I've pushed a fix.  In practice, when a user clicked "Attach INSPIRE Record" from the Dashboard, the code in `inspire-widget.html` calls code from `inspire.js` which makes a check `if (!inspire_ds.is_null(data) && data.status == 'success')` before showing the "Confirm" button.  A failed request will give null data so the "Confirm" button will be hidden.  However, clicking the "Confirm" button initiates the `/attach_information/` endpoint which makes another INSPIRE request, which could potentially fail even the first request was successful.  I think this must have happened a few months ago when the INSPIRE site was temporarily down and a user was able to finalise a record which ended up with null entries for the publication information.  Let me know if you approve this PR now, then I can merge.